src/MCPClient/lib/clientScripts/archivematicaUpdateSizeAndChecksum.py

```diff
@@ -89,7 +89,11 @@ def get_file_info_from_mets(shared_path, file_):
             relationship.related_event_identifier_value)
         if (not event) or (event.type != 'normalization'):
             continue
-        related_object_uuid = relationship.related_object_identifier_value
+        rel_obj_uuid = relationship.related_object_identifier_value
+        related_object_fsentry = mets.get_file(file_uuid=rel_obj_uuid)
+        if related_object_fsentry.use != 'preservation':
```

Comment:
I tested and this throws an error.

I think it is because the call to mets.get_file() is returning None where the file belongs to a fileGrp with use='deleted'.  Those files do not hvae a mets:FLocat element, maybe that is why get_file() returns None?

I changed this to the code here and it worked:
```
if related_object_fsentry is None:
            continue  # no fsentry for 'deleted' files 
```

Note sure if that is the best solution.

Reply:
@jhsimpson I think your solution probably works, but just in case it's possible for an original to be linked to something as a derivative that DOES have an "fsentry" in the METS but which does not have use="preservation", I've done the following instead (which will avoid the `AttributeError`):

    >>> if getattr(related_object_fsentry, 'use', None) != 'preservation':
    ...     continue